azurelinuxagent/common/utils/cryptutil.py

```diff
@@ -99,6 +109,13 @@ def decrypt_p7m(self, p7m_file, trans_prv_file, trans_cert_file, pem_file):
             except shellutil.CommandError as command_error:
                 logger.error("Failed to decrypt {0} (return code: {1})\n[stdout]\n{2}\n[stderr]\n{3}",
                     p7m_file, command_error.returncode, command_error.stdout, command_error.stderr)
+                # If the decryption fails, old version of openssl overwrite the output file(if exist) with empty data while
+                # new version of openssl does not overwrite the output file, So output file may contain old certs data.
```

Comment:
let's add the "new version" value to the comment, in case we ever need it in the future